modules/core/src/main/java/org/apache/ignite/internal/commandline/CommandHandler.java

```diff
@@ -1302,14 +1301,20 @@ private void baselinePrint0(VisorBaselineTaskResult res) {
 
         Map<String, VisorBaselineNode> srvs = res.getServers();
 
-        Optional<VisorBaselineNode> crdOpt = srvs.values().stream()
-            .min(Comparator.comparing(VisorBaselineNode::getOrder));
-
-        String crdStr = crdOpt.map(
-            crd -> "ConsistentId=" + crd.getConsistentId() + ", Order=" + crd.getOrder())
+        // 1) if we have a node with order == 1, then it's the CRD, otherwise
+        // 2) if we have a node with order == null, then CRD can't be evaluated, otherwise
+        // 3) the node with minimal order is CRD
+
+        String crdStr = srvs.values().stream()
+            // sort: 1, null, 2, 3, ...
+            .min(Comparator.comparing(node1 -> node1.getOrder() != null ? node1.getOrder() << 1 : 3))
+            // check for not null
+            .filter(node1 -> node1.getOrder() != null)
+            // format
+            .map(crd -> " (Coordinator: ConsistentId=" + crd.getConsistentId() + ", Order=" + crd.getOrder() + ")")
```

Comment:
This place of code is difficult to understanding.
1) What mean doubled order (node1.getOrder() << 1)?
2) Why null order in third position? (Integer.MAX_VALUE why not?)
3) After that you filtered all nodes with null order, but min() return only one... may by with nulled order node.
Need to rewrite.

Reply:
thanks @vldpyatkov. I've simplified code. Actually we can have 2 cases only
1)  all nodes  order != null  in case task launched on a new node =>  crd = min node
2) all nodes orde == null in case task launched on old versioned node  => crd can't be specified.